SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -0,0 +1,806 @@
+      }];
+}
+
+- (void)sdl_updateService:(UInt8)service
+        allowedTransports:(nullable NSArray<SDLTransportClassBox *> *)transportList
+         primaryAvailable:(BOOL)primaryTransportAvailable
+       secondaryAvailable:(BOOL)secondaryTransportAvailable
+    transportUpdatedBlock:(void (^)(SDLProtocol * _Nullable oldProtocol, SDLProtocol * _Nullable newProtocol))transportUpdatedBlock {
+
+    if (transportList == nil) {
+        // SDL Core did not send transport list for this service. Let's start it on Primary Transport for safe.
+        SDLLogW(@"Transport for service 0x%2X is not specified by the system, using primary transport", service);
+        transportList = @[@(SDLTransportClassPrimary)];
+    }
+
+    SDLTransportClass newTransport = SDLTransportClassInvalid;
+    // the list is in preferred order, so take a look from the beginning
+    for (SDLTransportClassBox *transport in transportList) {
+        if ([transport intValue] == SDLTransportClassSecondary && secondaryTransportAvailable) {
+            newTransport = SDLTransportClassSecondary;
+            break;
+        } else if ([transport intValue] == SDLTransportClassPrimary && primaryTransportAvailable) {
+            newTransport = SDLTransportClassPrimary;
+            break;
+        }
+    }
+
+    SDLTransportClass oldTransport = [self.assignedTransport[@(service)] intValue];
+
+    // update transport, and notify the change to start/stop/restart service
+    self.assignedTransport[@(service)] = @(newTransport);
+
+    if (oldTransport != newTransport) {
+        if (oldTransport != SDLTransportClassInvalid && newTransport != SDLTransportClassInvalid) {
+            SDLLogD(@"Restarting service 0x%X from %@ to %@ transport", service, [self sdl_getTransportClassName:oldTransport], [self sdl_getTransportClassName:newTransport]);
+        } else if (oldTransport != SDLTransportClassInvalid) {
+            SDLLogD(@"Stopping service 0x%X on %@ transport", service, [self sdl_getTransportClassName:oldTransport]);
+        } else if (newTransport != SDLTransportClassInvalid) {
+            SDLLogD(@"Starting service 0x%X on %@ transport", service, [self sdl_getTransportClassName:newTransport]);
+        }
+
+        transportUpdatedBlock([self sdl_getProtocolFromTransportClass:oldTransport],
+                              [self sdl_getProtocolFromTransportClass:newTransport]);
+    }
+}
+
+- (nullable SDLProtocol *)sdl_getProtocolFromTransportClass:(SDLTransportClass)transportClass {
+    switch (transportClass) {
+        case SDLTransportClassPrimary:   return self.primaryProtocol;
+        case SDLTransportClassSecondary: return self.secondaryProtocol;
+        default:                         return nil;
+    }
+}
+
+- (nullable NSString *)sdl_getTransportClassName:(SDLTransportClass)transportClass {
+    switch (transportClass) {
+        case SDLTransportClassPrimary:   return @"primary";
```

Comment:
Don't align the `return` sections, just put them one space after the `:`

Reply:
Fixed in commit c6d6d98423e3372a89ad1f52331bd5f1b8c65fd6.